pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/SetOperator.java

```diff
@@ -119,6 +119,7 @@ protected void constructRightBlockSet() {
           _rightRowSet.add(new Record(row));
         }
       }
+      sampleAndCheckInterruption();
```

Comment:
I think we should also have to call this method during `constructResultBlockSet`

Reply:
Done